pkg/plugins/user.go

```diff
@@ -112,20 +111,34 @@ func createUser(fs vfs.FS, u schema.User, console Console) error {
 			return errors.Wrap(err, "invalid uid defined")
 		}
 	} else {
-		// find an available uid if there are others already
 		all, _ := passwd.ParseFile(etcpasswd)
 		if len(all) != 0 {
-			usedUids := []int{}
-			for _, entry := range all {
-				uid, _ := strconv.Atoi(entry.Uid)
-				usedUids = append(usedUids, uid)
+			// Check if user is already in there to reuse the same UID as to not break existing permissions
+			existing := false
+			for name, values := range all {
+				if name == u.Name {
+					uid, err = strconv.Atoi(values.Uid)
+					if err != nil {
+						return errors.Wrap(err, "could not parse existing user id")
+					}
+					existing = true
+					break
+				}
 			}
-			sort.Ints(usedUids)
-			if len(usedUids) == 0 {
-				return errors.New("no new UID found")
+			// If it's not there, get a new UID
+			if !existing {
+				usedUids := []int{}
+				for _, entry := range all {
+					uid, _ := strconv.Atoi(entry.Uid)
+					usedUids = append(usedUids, uid)
+				}
+				sort.Ints(usedUids)
+				if len(usedUids) == 0 {
+					return errors.New("no new UID found")
+				}
+				uid = usedUids[len(usedUids)-1]
```

Comment:
I'm not sure where uid is used, if i'ts about line 133 then 132 always over writes it

Reply:
I did not touch this, was original code :D

But yes this is ok no? Gets the len of the used ids -1 to get an int and then adds 1 to it to get the next available

Could probably just be easier to do `uid = usedUids[len(usedUids)-1]` ??

Reply:
silly me, uid is used later on in 160